packages/flutter/lib/src/rendering/box.dart

```diff
@@ -471,6 +471,28 @@ class BoxParentData extends ParentData {
 /// ContainerRenderObjectMixin.
 abstract class ContainerBoxParentDataMixin<ChildType extends RenderObject> extends BoxParentData with ContainerParentDataMixin<ChildType> { }
 
+enum _IntrinsicDimension { minWidth, maxWidth, minHeight, maxHeight }
+class _IntrinsicDimensionsCacheEntry {
+  _IntrinsicDimensionsCacheEntry(this.dimension, this.argument);
+
+  final _IntrinsicDimension dimension;
+  final double argument;
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
```

Comment:
How do we decide whether to include this check?  In this case, I suspect it doesn't make it faster.